integrations/linear/src/webhooks.ts

```diff
@@ -0,0 +1,341 @@
+import {
+  ExternalSource,
+  ExternalSourceTrigger,
+  HandlerEvent,
+  IntegrationTaskKey,
+  Logger,
+} from "@trigger.dev/sdk";
+import {
+  LinearWebhooks,
+  LINEAR_WEBHOOK_SIGNATURE_HEADER,
+  LINEAR_WEBHOOK_TS_FIELD,
+  WebhookPayload,
+  DeletePayload,
+  Webhook,
+} from "@linear/sdk";
+import { z } from "zod";
+import * as events from "./events";
+import { Linear, LinearRunTask } from "./index";
+import {
+  WebhookCreateInput,
+  WebhookUpdateInput,
+  WebhooksQueryVariables,
+} from "@linear/sdk/dist/_generated_documents";
+import { WebhookPayloadSchema } from "./schemas";
+
+export const WebhookResourceTypeSchema = z.union([
+  z.literal("Attachment"),
+  z.literal("Comment"),
+  z.literal("Cycle"),
+  z.literal("Issue"),
+  z.literal("IssueLabel"),
+  z.literal("Project"),
+  z.literal("ProjectUpdate"),
+  z.literal("Reaction"),
+]);
+export type WebhookResourceType = z.infer<typeof WebhookResourceTypeSchema>;
+
+export const WebhookActionTypeSchema = z.union([
+  z.literal("create"),
+  z.literal("remove"),
+  z.literal("update"),
+]);
+export type WebhookActionType = z.infer<typeof WebhookActionTypeSchema>;
+
+type DeleteWebhookParams = {
+  id: string;
+};
+
+type UpdateWebhookParams = {
+  id: string;
+  input: WebhookUpdateInput;
+};
+
+// TODO: types
+const withoutFunctions = <T>(obj: T): T => {
+  return JSON.parse(JSON.stringify(obj), (key, value) => {
+    if (typeof value === "function") {
+      return undefined;
+    }
+    return value;
+  });
+};
+
+export class Webhooks {
+  runTask: LinearRunTask;
+
+  constructor(runTask: LinearRunTask) {
+    this.runTask = runTask;
+  }
+
+  create(
+    key: IntegrationTaskKey,
+    params: WebhookCreateInput
+    // TODO: tidy up return type
+  ): Promise<Omit<WebhookPayload, "webhook"> & { webhook: Webhook | undefined }> {
+    return this.runTask(
+      key,
+      async (client, task, io) => {
+        const payload = await client.createWebhook(params);
+        return withoutFunctions({
+          ...payload,
+          webhook: await payload.webhook,
+        });
```

Comment:
Problem: The Linear SDK likes to use getters. Also, `JSON.stringify()` will expose private properties that could be hidden via the return type, so it's not possible to remove functions generally. I could add a deep function remover without stringification instead.

Reply:
It looks like you might have solved this problem now?

Reply:
Yes, thought I'd marked this as resolved already! Extended the JSON approach and added a type to keep it simple.